docs/scope-and-objectives.md

```diff
@@ -0,0 +1,139 @@
+- To configure a machine at any time other than create or upgrade.
+- To duplicate functionality that exists or is coming to other tooling, e.g., updating kubelet configuration (c.f. dynamic kubelet configuration), or updating apiserver, controller-manager, scheduler configuration (c.f. component-config effort) after the cluster is deployed.
+
+## Requirements
+
+#### Foundation
+
+- Cluster API MUST be able to be deployed and run on Kubernetes.
+
+- Cluster API MUST provide isolation between providers.
+
+- Cluster API MUST support multiple infrastructure providers, including both on-prem and cloud providers.
+
+- Cluster API MUST be able to manage groups/sets of Kubernetes nodes that support scaling and orchestrated upgrades.
+
+- Cluster API, through a single installation, MUST be able to manage (create, scale, upgrade, destroy) clusters on multiple providers.
+
+- Cluster API SHOULD support all operating systems in scope for Kubernetes conformance.
+
+#### User Experience
+
+- Cluster API MUST be able to display the versions of Kubernetes and related components of a Node that it manages.
+
+- Cluster API MUST be able to provide sufficient information for a consumer to access a provisioned Kubernetes cluster’s API server.
+
+#### Organization
+
+- Cluster API MUST NOT have code specific to a particular provider within the main repository, defined as sigs.k8s.io/cluster-api.
+
+#### Validation
+
+- Cluster API provider implementations SHOULD validate configuration data.
+
+- Cluster API MUST test integrations that it claims will work.
+
+- Cluster API MUST deploy clusters that pass Kubernetes conformance tests.
+
+- Cluster API MUST document how to bootstrap itself and MAY provide tooling to do so.
+
+- Cluster API MUST document requirements for machine images and SHOULD reference default tools to prepare machine images.
+
+- Cluster API MUST have conformance tests to ensure that Cluster API and a given provider fulfill the expectations of users/automation.
+
+#### Extension
+
+- Cluster API SHOULD allow default implementations to be pluggable/extensible.
+   - Cluster API SHOULD offer default implementations for generic operations.
+   - Users MAY use different provider implementations for different sets of operations.
+
+- 🔌 Cluster API MUST be able to provide information as to which versions of Kubernetes it can install.
+
+- 🔌 Cluster API MUST be able to bootstrap a new Kubernetes control plane.
+
+- 🔌 Cluster API MUST provide a generic image lookup mechanism that can filter on characteristics such as kubernetes version, container runtime, kernel version, etc.
+
+- 🔌 Cluster API MUST be able to provide a consistent (across providers) way to prepare, configure, and join nodes to a cluster.
+
+- 🔌 Cluster API MUST be able to scale a self-provisioned control plane up from and down to one replica.
+
+- 🔌 Cluster API MAY be able to scale an external control plane up from and down to one replica.
```

Comment:
No - this applies to using Cluster API to create a `Cluster` where the provider is GKE / EKS / etc.

Reply:
But is that case, the control plane is not provisioned by clusterAPI, right? I mean CAPI won't have any control on the bootstrap process.

Reply:
I see. Makes sense. I was under the impression we wanted to offer some control on the control plane configuration (at least this was reflected in the use case document). But if we aim also externally provided control planes, this cannot be consistently provided across providers.

Reply:
We probably will look into offering control over the control plane configuration. It may not apply to all Kubernetes clusters you can potentially install using Cluster API. Either way, I'm not sure that's relevant to this optional requirement about scaling external control planes 😄.

Reply:
No it doesn't. It is more a reflexion that came from your comment "we think it's probably desirable to have CAPI manage as many clusters as possible". That said. I still think it's a valid observation.

Reply:
No it doesn't. It is more a reflexion that came from your comment "we think it's probably desirable to have CAPI manage as many clusters as possible". That said. I still think it's a valid observation.

Reply:
No it doesn't. It is more a reflexion that came from your comment "we think it's probably desirable to have CAPI manage as many clusters as possible". That said. I still think it's a valid observation

Reply:
No it doesn't. It is more a reflexion that came from your comment "we think it's probably desirable to have CAPI manage as many clusters as possible". That said. I still think it's a valid observation

Reply:
I'm not sure this statement actually belongs in CAPI.  This capability is provided by a provider for those external control plane.  I think this statement is here because of a desire to call out this possibility out, but CAPI will probably not have any special code to allow this functionality.

Reply:
IIRC this item has been discussed in the document and over the extra Zoom meetings last week, we did have only the requirement above, but someone noted that Cluster API might have the ability to scale providers that cannot be explicitly managed.

Let's say for example we have an external provider called Foo. If Foo offers Kubernetes as a service, they might also offer ability for example to scale up a Control Plane from a single node to HA (just as an example). If Cluster API offers extension points that any provider can fulfill to scale up and down a Control Plane, Foo provider can fill those extension points with actual implementations and check the box on the feature list. 

Does the example clarify the purpose of the requirement? To be clear, I'm not advocating for keeping it or not, just trying to provider as much information as possible.

Reply:
@pablochacin @sflxn Does the explanation above resolve this conversation?

Reply:
@vincepri Thanks for the clarification. I still agree with @sflxn. I'm unconvinced this should be requirement. I would propose to leave it as out of scope initially (as other uses cases) what doesn't prevent us for considering it - if I understand "out of scope" correctly - but certainly would allow us to focus in managed control planes. All that said, I'm not proposing considering an strict  non-goal, either.

Reply:
@vincepri @pablochacin I'm fine with leaving it in or taking it out.

Reply:
Took it out for now, we can revisit later.